Danki2/Assets/Scripts/Actor/Actor.cs

```diff
@@ -13,27 +13,22 @@ protected virtual void Start()
     {
         _statsManager = new StatsManager(baseStats);
         _effectTracker = new EffectTracker(this);
+        _statsManager.RegisterPipe((stat, value) => _effectTracker.ProcessStat(stat, value));
```

Comment:
The effect tracker needs to call `ResetCache()` on the stats manager if an effect expires or if a new effect is added. I think it would be better to pass the stats manager through to the effect tracker and have it register itself in its constructor, rather than the actor registering the effect tracker for it here. Then the effect tracker can also hold on to the reference to the stats manager to reset the cache at the right times